Example-Audit-REXX/RXAUDIT1.rexx

```diff
@@ -0,0 +1,2478 @@
+/*        -I indicate running out of ISV REXX environment,             */
+/*           default is TSO/E                                          */
+/*        -V turn on additional verbose JSON tracing                   */
+/*                                                                     */
+/*        EX 'HWI.HWIREST.REXX(RXAUDIT1))'                             */
+/*            '-D HWI.AUDIT -I'                                        */
+/*            data set is HWI.AUDIT, run out of ISV REXX,              */
+/*            if successful will create HWI.AUDIT(LOCAL)               */
+/*                                                                     */
+/* DEPENDENCIES                                                        */
+/*     none.                                                           */
+/*                                                                     */
+/*    NOTES:                                                           */
+/* No recovery logic has been supplied in this sample.                 */
+/*                                                                     */
+/*    REFERENCE:                                                       */
+/*        See the z/OS MVS Programming: Callable Services for          */
+/*        High-Level Languages publication for more information        */
+/*        regarding the usage of HWIREST and JSON Parser APIs.         */
+/*                                                                     */
+/*                                                                     */
+/* END OF SPECIFICATIONS  * * * * * * * * * * * * * * * * * * * * * * **/
+
+MACLIB_DATASET = 'SYS1.MACLIB'
+
+TEMP_DATASET = '' /* required input via -D */
+MEMBER_NAME = 'LOCAL' /* default CPC */
+
+TRUE = 1
+FALSE = 0
+
+bRC = 0          /* HWIHOST rc */
+HWIHOST_ON = FALSE
+PARSER_INIT = FALSE
+ISVREXX = FALSE  /* default to TSO/E, enable via -I */
+VERBOSE = FALSE  /* JSON parser specific, enabled via -V */
+localCPC = TRUE  /* default to lpars on the LOCAL CPC, change via -C */
+
+/* Complex Properties constants */
+storageCentralAllocProp = 'storage-central-allocation'
+storageExpandedAllocProp = 'storage-expanded-allocation'
+
+/*********************************/
+/* Get program args              */
+/*********************************/
+parse arg argString
+if GetArgs(argString) <> 0 then
+   exit -1
+
+/*********************************/
+/* Before going  much further,   */
+/* ensure the data set specified */
+/* actually exists               */
+/*********************************/
+if VerifyDataSet(TEMP_DATASET) <> 0 then
+  exit fatalErrorAndCleanup('** specified data set does not exit **')
+
+if localCPC then
+  say 'Obtaining LPARs on LOCAL CPC'
+else
+  say 'Starting audit of LPARs on CPC:'||CPCname
+
+if ISVREXX then
+  do
+    say 'Running in an ISV REXX environment'
+    bRC = hwihost("ON")
+    say 'HWIHOST("ON") return code is ; 'bRC
```

Comment:
Shouldn't this be 'HWIHOST("OFF") return code is ; ' || bRC ?

Reply:
not necessarily, the || (concatenation) bars guarantee there isn't going to be any extra space between '; ' and the bRC but in general, if you're okay chancing it, you can just put the variable after the string, but given everywhere else i used the '||' i'll fix this up